core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -421,7 +433,7 @@ private ManifestFile filterManifestWithDeletedFiles(
                 entry -> {
                   F file = entry.file();
                   boolean markedForDelete =
-                      deletePaths.contains(file.path())
+                      fileIsDeleted(file, manifest)
```

Comment:
Open to better names here, I was more so creating a helper method to group the position based and the path based check, since there's a lot of conditions here.

Reply:
Also just noticed for the pos based comparison we may not be really optimizing much since  for the duplicate file detection case we do another string comparison on line 457. So we really only remove one path string comparison instead of all of them in the current implementation.